### PR TITLE
Remove DoFHandlerType from dof_handler_policy.h/.cc

### PR DESCRIPTION
**Pull request by peterrum**
... part of #10333.

### CODE REVIEW

File: source/dofs/dof_handler_policy.cc [hunk 1]
```diff
@@ -2990,9 +2988,12 @@ namespace internal
...
-          typename DoFHandlerType::cell_iterator cell =
-                                                   dof_handler.begin(level),
-                                                 endc = dof_handler.end(level);
+          typename DoFHandler<dim, spacedim>::cell_iterator cell =
+                                                              dof_handler.begin(
+                                                                level),
+                                                            endc =
+                                                              dof_handler.end(
+                                                                level);
           for (; cell != endc; ++cell)
```
Comment (kronbichler) [inline source/dofs/dof_handler_policy.cc:2997]: Can we use this:
```suggestion
          for (const auto &cell : dof_handler.cell_iterators_on_level(level))
```

**Comment by peterrum** [comment]
@kronbichler Done!

File: source/dofs/dof_handler_policy.cc [hunk 2]
```diff
@@ -4064,7 +4041,7 @@ namespace internal
               {
                 std::vector<dealii::types::global_dof_index> local_dof_indices;
 
-                typename DoFHandlerType::level_cell_iterator
+                typename DoFHandler<dim, spacedim>::level_cell_iterator
                   cell = dof_handler->begin(level),
                   endc = dof_handler->end(level);
```
Comment (kronbichler) [inline source/dofs/dof_handler_policy.cc:4046]: I the second I pressed `approve` I saw these objects. Can we convert the place here and a few more below to range-based for loops as well?

Comment (peterrum) [reply source/dofs/dof_handler_policy.cc]: Done!

**Pull request merged**